internal/manifest/constants.go

```diff
@@ -3,32 +3,61 @@
 
 package manifest
 
-// TrustedMeasurement contains the expected launch digest and is injected at build time.
-var TrustedMeasurement = "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
-
-// Default returns a default manifest.
-func Default() Manifest {
-	return Manifest{
-		ReferenceValues: ReferenceValues{
-			TrustedMeasurement: HexString(TrustedMeasurement),
-		},
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/edgelesssys/contrast/node-installer/platforms"
+)
+
+// Default returns a default manifest with reference values for the given platform.
+func Default(platform platforms.Platform) (*Manifest, error) {
+	refValues := setReferenceValuesIfUninitialized()
+
+	mnfst := Manifest{}
+	switch platform {
+	case platforms.AKSCloudHypervisorSNP:
+		return &Manifest{
+			ReferenceValues: ReferenceValues{
+				AKS: refValues.AKS,
+			},
+		}, nil
+	case platforms.RKE2QEMUTDX, platforms.K3sQEMUTDX:
+		return &Manifest{
+			ReferenceValues: ReferenceValues{
+				BareMetalTDX: refValues.BareMetalTDX,
+			},
+		}, nil
 	}
+	return &mnfst, nil
 }
 
-// DefaultAKS returns a default manifest with AKS reference values.
-func DefaultAKS() Manifest {
-	mnfst := Default()
-	mnfst.ReferenceValues.SNP = SNPReferenceValues{
-		MinimumTCB: SNPTCB{
-			BootloaderVersion: toPtr(SVN(3)),
-			TEEVersion:        toPtr(SVN(0)),
-			SNPVersion:        toPtr(SVN(8)),
-			MicrocodeVersion:  toPtr(SVN(115)),
-		},
+// DefaultPlatformHandler is a short-hand for getting the default runtime handler for a platform.
+func DefaultPlatformHandler(platform platforms.Platform) (string, error) {
+	mnf, err := Default(platform)
+	if err != nil {
+		return "", fmt.Errorf("generating manifest: %w", err)
 	}
-	return mnfst
+	return mnf.RuntimeHandler(platform)
 }
 
-func toPtr[T any](t T) *T {
-	return &t
+// EmbeddedReferenceValues returns the reference values embedded in the binary.
+func EmbeddedReferenceValues() ReferenceValues {
+	return setReferenceValuesIfUninitialized()
+}
+
+// EmbeddedReferenceValuesIfUninitialized returns the reference values embedded in the binary, initializing
+// the global state if it is not yet initialized.
+func setReferenceValuesIfUninitialized() ReferenceValues {
+	if embeddedReferenceValues == nil {
+		// If we're here, this is the first time this function is called, and the global state is not
+		// yet initialized. So let's unmarshal the embedded reference values. This does not need to be
+		// locked, as embeddedReferenceValues has a fixed value at compile time, making this idempotent.
+		if err := json.Unmarshal(EmbeddedReferenceValuesJSON, &embeddedReferenceValues); err != nil {
+			fmt.Printf("Failed to unmarshal embedded reference values: %s\n", err)
+			os.Exit(1)
```

Comment:
os.Exit is only allowed in top-level functions (main package).

Reply:
This would just add additional code complexity as we'd have to bubble up and handle the error everywhere. What about using an init, as @burgerdev suggested? I just don't see the advantage with this

Reply:
No, no init. We even run a linter to prevent you from using it.

If you cannot live without global state, just use panic here instead of os.Exit and cover it with a unit test.

Reply:
We could also just keep it simple and parse the JSON every time `EmbeddedReferenceValues` is called - which is not going to be that often, in one CLI run? And panic sounds fine since we're dealing with constant, predictable data and should fail loudly in case it is not.

Reply:
Ok, will do